File: Eglantine/Data/Events/FrontYard_events.lua
function enterFrontYard()
	Event:StopLoopingSoundEffects()
	Event:PlaySoundLooping("naturalwind", .3, 0, 0)
	checkForBloom()
end

function checkForBloom()
	if Event:PlayerHasItem("Photograph") and not GameState.FrontYardFlowerPicked then
		Event:EnableInteractable("Eglantine")
	end
end

function pickFrontYardEglantine()
	if(Event:UsingItem("Scissors")) then
		pickup("Eglantine")
		GameState.FrontYardFlowerPicked = true
	end
end

function leaveFrontYard()
	Event:StopLoopingSoundEffects()
	Event:PlaySoundLooping("lowwind", .25, 0, 0)
end

File: Eglantine/Scripts/RoomEvents/FrontYardEvents.cs
using System;
using System.Collections.Generic;
using ObjectivelyRadical.Scheduler;

namespace Eglantine.Engine
{
	public static partial class GameEvents
	{

		public static IEnumerator<ScriptPauser> enterFrontYard()
		{
			EventManager.Instance.StopLoopingSoundEffects();
			EventManager.Instance.PlaySoundLooping("naturalwind", .3f, 0f, 0f);
			if(EventManager.Instance.PlayerHasItem("Photograph") && !GameState.Instance.FrontYardFlowerPicked)
				EventManager.Instance.EnableInteractable("Eglantine");

			yield return null;
		}

		public static IEnumerator<ScriptPauser> pickFrontYardEglantine ()
		{
			if (EventManager.Instance.UsingItem ("Scissors"))
			{
				pickup ("Eglantine");

				yield return waitUntil("Player stopped");
				GameState.Instance.FrontYardFlowerPicked = true;
			}

			yield return null;
		}

		public static IEnumerator<ScriptPauser> leaveFrontYard()
		{
			EventManager.Instance.StopLoopingSoundEffects();
			EventManager.Instance.PlaySoundLooping("lowwind", .25f, 0, 0);

			yield return null;
		}


// --LOOK EVENTS-- //

		public static IEnumerator<ScriptPauser> lookAtFrontDoor()
		{
			EventManager.Instance.ShowMessage("What a creepy looking place...");

			yield return null;
		}

		public static IEnumerator<ScriptPauser> lookAtFrontYardFlower()
		{
			EventManager.Instance.ShowMessage("Was that flower there before...?");
			
			yield return null;
		}

// --PICK UP EVENTS-- //

// --DOORS-- //
		public static IEnumerator<ScriptPauser> useFrontYardDoor()
		{
			door("Door", "Foyer", "FrontDoor");
			yield return null;
		}

	}
}

